cmd/internal/cli/delete.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/sylabs/scs-library-client/client"
+	"github.com/sylabs/singularity/docs"
+	"github.com/sylabs/singularity/internal/app/singularity"
+	scs "github.com/sylabs/singularity/internal/pkg/remote"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+	"github.com/sylabs/singularity/internal/pkg/util/interactive"
+	"github.com/sylabs/singularity/pkg/cmdline"
+)
+
+func init() {
+	cmdManager.RegisterCmd(deleteImageCmd)
+	cmdManager.RegisterFlagForCmd(&deleteImageArchFlag, deleteImageCmd)
+	cmdManager.RegisterFlagForCmd(&deleteImageTimeoutFlag, deleteImageCmd)
+	cmdManager.RegisterFlagForCmd(&deleteLibraryURIFlag, deleteImageCmd)
+}
+
+var deleteImageArch string
+var deleteImageArchFlag = cmdline.Flag{
+	ID:           "deleteImageArchFlag",
+	Value:        &deleteImageArch,
+	DefaultValue: "",
+	Name:         "arch",
+	ShortHand:    "A",
+	Required:     true,
+	Usage:        "specify requested image arch",
+	EnvKeys:      []string{"ARCH"},
+}
+
+var deleteImageTimeout int
+var deleteImageTimeoutFlag = cmdline.Flag{
```

Comment:
Is this something that should be changed with a flag? (I think thats okay) but I do suggest making the flag hidden. Because this is something that should not be needing to be changed often (or at all). Afaik. 

I honestly think a flag like this would also be useful for the `push`.